homeassistant/components/weather/__init__.py

```diff
@@ -322,6 +326,55 @@ def __init_subclass__(cls, **kwargs: Any) -> None:
                     cls.__name__,
                     report_issue,
                 )
+        if any(method in cls.__dict__ for method in ("_attr_forecast", "forecast")):
```

Comment:
We should only warn if the class has overridden at least one of "_attr_forecast" or "forecast" but does not implement any of the new async methods, otherwise we will warn also for upgraded classes.

Reply:
This was marked as resolved, but I don't see that it's fixed?

Reply:
Right. I misunderstood what you meant.
Will update